### PR TITLE
Feature/set up associations

### PR DESCRIPTION
**Pull request by adel-gu**
# Changes implemented in this PR:

- Create a class `Classroom` with the `@label` instance variable, with Setter and getter for `@label`.

- Create the `has-many/belongs-to` relationship between `Classroom `and `Student`. 

- Create a class `Book` with `@title` and `@author` instance variables with their Setters and getters.

- Create a class Rental with the `@date` instance variable with its Setter and getter.

- Create the `many-to-many` relationship between `Person` and `Book` using `Rental`.

### CODE REVIEW

File: models/book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Book
+  # Getter and Setters
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (rustam-tolipov) [inline models/book.rb:11]: - [x] A method for adding rentals should be present ([more info](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)). Kindly add the `add_rental` method in your Book class :)

File: models/person.rb [hunk 2]
```diff
@@ -3,21 +3,20 @@
...
+  attr_accessor :name, :age
+
   # Constructor
   def initialize(age, name = 'Unkown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (rustam-tolipov) [inline models/person.rb:18]: - [x] It's great you initialized your `rentals`. However, I couldn't find any method for adding rentals in your Person class. Make sure you have created a month for adding rentals here 😉

File: models/person.rb [hunk 3]
```diff
@@ -3,21 +3,20 @@
 require_relative 'trimmer_decorator'
 
 class Person < Nameable
+  # getters and setters
+  attr_reader :id, :rentals
```
Comment (rustam-tolipov) [inline models/person.rb]: [OPTIONAL] You are getting `:rentals` with the getters method. But you also need a setters method for `:rentals`. Or you have a free chance of making both in one place by using `attr_accessor `

File: models/book.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Book
+  # Getter and Setters
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (rustam-tolipov) [inline models/book.rb]: [OPTIONAL] You are getting `:rentals` with the getters method. But you also need a setters method for `:rentals`. Or you have a free chance of making both in one place by using `attr_accessor `

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @adel-gu,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Kindly remove the Microverse logo from the readme file.

_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by adel-gu** [comment]
Hi @Rustamxon7 👋

Thank you very much👏👏, the suggestions were very helpful ✨.

**Review by ha-manel — APPROVED** [review]
# STATUS: APPROVED ✅

## Hi @adel-gu👋

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉**

I like that you implemented even the optional suggestions 😊 Keep rocking 🚀 



## Highlights:
* Well-written PR description ✔️ 
* All tests passed ✔️ 
* Associations are setup correctly 💯 


------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**